### PR TITLE
Meeting view implementation

### PR DESCRIPTION
**Pull request by bierik**
Made meeting view implementation as described in https://basecamp.com/2768704/projects/7577904/todos/170675783.
Updated builders for generating meeting related documents.
Updated translations for shorter protocol actions.

Depends on https://github.com/4teamwork/plonetheme.teamraum/pull/299

### CODE REVIEW

File: opengever/meeting/browser/meetings/templates/meeting.pt [hunk 1]
```diff
@@ -12,69 +12,91 @@
...
-                               var msg_unexpected_error = '${view/msg_unexpected_error}';"></script>
-
-          <h2 i18n:translate="">Properties</h2>
-          <table class="vertical listing">
+  <metal:content-core fill-slot="content-core">
+      <metal:content-core define-macro="content-core"
+                          tal:define="meeting view/model;
+                                      unscheduled_proposals view/unscheduled_proposals">
+      <div id="opengever_meeting_meeting">
+        <script tal:content="string:var js_update_order_url = '${view/url_update_agenda_item_order}'
```
Comment (phgross) [inline opengever/meeting/browser/meetings/templates/meeting.pt]: I know it's no really part of this change, but could you change that implementation to using the data attribute.

You'll find a simple example in the notification viewlet: 
 - https://github.com/4teamwork/opengever.core/blob/master/opengever/activity/viewlets/notification.pt#L6
 - https://github.com/4teamwork/opengever.core/blob/master/opengever/activity/viewlets/notification.py#L50
 - https://github.com/4teamwork/opengever.core/blob/master/opengever/activity/browser/resources/notifications.js#L13

File: opengever/meeting/browser/meetings/templates/meeting.pt [hunk 2]
```diff
@@ -12,69 +12,91 @@
...
+            </table>
+            <metal:use use-macro="context/@@meeting-macros/workflow_actions" />
+          </fieldset>
+          <fieldset class="excerpts" tal:define="excerpts view/manually_generated_excerpts">
+            <legend i18n:translate="">Excerpts</legend>
+            <tal:condition tal:condition="excerpts">
+              <ul>
+                <li tal:repeat="item excerpts">
+                  <a href="" tal:attributes="href item/absolute_url;
+                                             title python: getattr(item, 'document_date', None) and item.document_date.strftime('%d.%m.%Y') or '';
```
Comment (phgross) [inline opengever/meeting/browser/meetings/templates/meeting.pt]: I would drop the `title` definition displaying the `document_date` makes not much sense.

File: opengever/meeting/model/agendaitem.py [hunk 3]
```diff
@@ -89,7 +89,12 @@ def get_title(self):
         return u"{} {}".format(self.number, title)
 
     def get_css_class(self):
-        return "paragraph" if self.is_paragraph else ""
+        css_classes = []
+        if self.is_paragraph:
+            css_classes.append("paragraph")
+        if self.has_submitted_documents():
```
Comment (phgross) [inline opengever/meeting/model/agendaitem.py]: Whats about a proposal who does not contain submitted documents, but has an excerpt ?

File: opengever/meeting/model/meeting.py [hunk 4]
```diff
@@ -211,10 +211,13 @@ def get_date(self):
         return api.portal.get_localized_time(datetime=self.start)
 
     def get_start_time(self):
-        return self._get_localized_time(self.start)
```
Comment (phgross) [inline opengever/meeting/model/meeting.py]: Why is this change necessary? the `_get_localized_time` method use `api.portal.get_localized_time(datetime=date, time_only=True)` this should work well...

File: opengever/meeting/tests/test_meeting_view.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        self.attachement = create(Builder('document').attach_file_containing(u"attachement", u"attachement.docx").within(self.dossier))
+
+        container = create(Builder('committee_container'))
+        self.committee = create(Builder('committee').within(container))
+
+        self.sablon_template = create(
+            Builder('sablontemplate')
+            .with_asset_file('excerpt_template.docx'))
+
+        self.preprotocol = create(Builder('document').attach_file_containing(u"preprotocol", u"preprotocol.docx").within(self.dossier))
```
Comment (phgross) [inline opengever/meeting/tests/test_meeting_view.py]: `ftw.builder` provide an easy way to take care to the line length, and make it much more readable:

``` python
    self.preprotocol = create(Builder('document')
                              .attach_file_containing(u"preprotocol",
                                                      u"preprotocol.docx")
                              .within(self.dossier))
```

File: opengever/meeting/tests/test_meeting_view.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    def test_participants_listing_no_secretary_must_not_raise(self, browser):
+        Meeting.query.all()[0].secretary = None
+        transaction.commit()
+        browser.login().open(MeetingList.url_for(self.committee, self.meeting))
+        self.assertEquals([''], browser.css("#meeting_secretary").text)
+
+    @browsing
+    def test_participants_listing_participants_is_existing(self, browser):
+        browser.login().open(MeetingList.url_for(self.committee, self.meeting))
+        self.assertEquals(['Peter Meter Hans Besen Roland Kuppler'], browser.css("#meeting_participants").text)
```
Comment (phgross) [inline opengever/meeting/tests/test_meeting_view.py]:
```python
self.assertEquals(['Peter Meter Hans Besen Roland Kuppler'], 
                             browser.css("#meeting_participants").text)
```

**Comment by phgross** [comment]
@bierik nice work just some adjustments needed:
 - see my comments
 - add a screenshot to the PR description
 - Fix tests, they are currently failing

**Comment by phgross** [comment]
@bierik ich habe das ganze noch kurz lokal getestet, sieht schon mal toll aus, folgende offene Punkte sind mir noch aufgefallen:

- `Attachments` und `Generated proposal excerpt` ist nicht übersetzt
- None Geschäfte A
- Das Ausklappen von Expandable works not correct
- Schliessen Icon fehlt
- Alle buttons sind immer möglich, müsste wohl am Status der jeweiligen Sitzung angepasst werden. Ich schlage vor das wir dies in einem separate PR angehen.
- Das Verschieben von Traktanden ist möglich obwohl Sitzung geschlossen, anschliessend gibt es einen Fehler)
- Der Abstand vom letzten zum Footer ist sehr klein, lässt sich da noch was machen.

**Comment by bierik** [comment]
@phgross My parts are done. Pls have a look.

**Comment by deiferni** [comment]
@phgross @bierik 
i hijacked this branch to:
- rebase onto master and actually push :stuck_out_tongue_closed_eyes: 
- eliminate a duplicate method

I also added a small changelog entry.

Unless tests dont pass i think it's ready now! :ship:

**Comment by phgross** [comment]
:+1:
